### PR TITLE
Handle project status through message log

### PR DESCRIPTION
**Pull request by JohnThomson**
Also adds core of TeamCollectionDialog

---
This change is [Reviewable](https://reviewable.io/reviews/bloombooks/bloomdesktop/4243)

### CODE REVIEW

**Review by andrew-polk — CHANGES_REQUESTED** [review]
> Reviewed 31 of 32 files at r1.
> *[Reviewable](https://reviewable.io/reviews/bloombooks/bloomdesktop/4243)* status: 31 of 32 files reviewed, 12 unresolved discussions (waiting on @JohnThomson)

___
*[src/BloomBrowserUI/teamCollection/TeamSettingsDialog.tsx, line 14 at r1](https://reviewable.io/reviews/bloombooks/bloomdesktop/4243#-MUojqGwEUIFIYZOzx_d:-MUojqGwEUIFIYZOzx_e:b-jeya46) ([raw file](https://github.com/bloombooks/bloomdesktop/blob/0e6b56dfb7ed8a9cebe1400125e3134cd527cdfb/src/BloomBrowserUI/teamCollection/TeamSettingsDialog.tsx#L14)):*
> ```
> TeamCollection.DialogTitle
> ```
We already have TeamCollection.TeamCollection
___
*[src/BloomExe/TeamCollection/NewBookEventArgs.cs, line 11 at r1](https://reviewable.io/reviews/bloombooks/bloomdesktop/4243#-MUox0RD5-uVCs5SXPSH:-MUox0RD5-uVCs5SXPSI:b-5m350p) ([raw file](https://github.com/bloombooks/bloomdesktop/blob/0e6b56dfb7ed8a9cebe1400125e3134cd527cdfb/src/BloomExe/TeamCollection/NewBookEventArgs.cs#L11)):*
> ```
> BookStateChangeEventArgs
> ```
BookStatusChangeEventArgs?
___
*[src/BloomExe/TeamCollection/TeamCollectionApi.cs, line 71 at r1](https://reviewable.io/reviews/bloombooks/bloomdesktop/4243#-MUopVJT0KOr7P-n3En7:-MUopVJT0KOr7P-n3En8:b-57dlby) ([raw file](https://github.com/bloombooks/bloomdesktop/blob/0e6b56dfb7ed8a9cebe1400125e3134cd527cdfb/src/BloomExe/TeamCollection/TeamCollectionApi.cs#L71)):*
> ```Smalltalk
> 			ReloadProjectAction?.Invoke();
> 			request.PostSucceeded();
> 		}
> ```

Thinking out loud...
Since more and more things are moving toward react dialogs, etc., should we consider moving HandleCloseDialog and HandleReload to something outside of Team Collections?
___
*[src/BloomExe/TeamCollection/TeamCollectionDialog.Designer.cs, line 4 at r1](https://reviewable.io/reviews/bloombooks/bloomdesktop/4243#-MUokszF47odgxy2oq3m:-MUokszF47odgxy2oq3n:bfei9we) ([raw file](https://github.com/bloombooks/bloomdesktop/blob/0e6b56dfb7ed8a9cebe1400125e3134cd527cdfb/src/BloomExe/TeamCollection/TeamCollectionDialog.Designer.cs#L4)):*
> ```Smalltalk
> namespace BloomTests.TeamCollection
> {
> 	partial class TeamCollectionDialog
> ```

Unless I'm mistaken, this is just a form which holds a single ReactControl.
Should this be a more generic class? Like ReactDialog?
___
*[src/BloomExe/TeamCollection/TeamCollectionManager.cs, line 74 at r1](https://reviewable.io/reviews/bloombooks/bloomdesktop/4243#-MUp2bcVAwfZg7mcIpw1:-MUp2bcVAwfZg7mcIpw2:bewhzo) ([raw file](https://github.com/bloombooks/bloomdesktop/blob/0e6b56dfb7ed8a9cebe1400125e3134cd527cdfb/src/BloomExe/TeamCollection/TeamCollectionManager.cs#L74)):*
> ```
> var tcLogPath = GetTcLogPathFromLcPath(_localCollectionFolder);
> ```
never used
___
*[src/BloomExe/TeamCollection/TeamCollectionMessage.cs, line 94 at r1](https://reviewable.io/reviews/bloombooks/bloomdesktop/4243#-MUotF-Q0GlaTB_lxHxw:-MUotF-R2HRiuoF5AaG0:bui0vzu) ([raw file](https://github.com/bloombooks/bloomdesktop/blob/0e6b56dfb7ed8a9cebe1400125e3134cd527cdfb/src/BloomExe/TeamCollection/TeamCollectionMessage.cs#L94)):*
> ```
> Displayed error log
> ```
This gets displayed in non-error conditions, sometimes, doesn't it?
How about just "Displayed log"?
___
*[src/BloomExe/TeamCollection/TeamCollectionMessage.cs, line 96 at r1](https://reviewable.io/reviews/bloombooks/bloomdesktop/4243#-MUotO0N5cfKeUh9BAst:-MUotO0N5cfKeUh9BAsu:bz6cnrx) ([raw file](https://github.com/bloombooks/bloomdesktop/blob/0e6b56dfb7ed8a9cebe1400125e3134cd527cdfb/src/BloomExe/TeamCollection/TeamCollectionMessage.cs#L96)):*
> ```
> Repaired conflict
> ```
We didn't really repair anything did we?
How about "Displayed conflict notification"?
___
*[src/BloomTests/TeamCollection/TeamCollectionDialog.cs, line 20 at r1](https://reviewable.io/reviews/bloombooks/bloomdesktop/4243#-MUolXdO6Fl8QqHLtXi9:-MUolXdO6Fl8QqHLtXiA:b-83cykv) ([raw file](https://github.com/bloombooks/bloomdesktop/blob/0e6b56dfb7ed8a9cebe1400125e3134cd527cdfb/src/BloomTests/TeamCollection/TeamCollectionDialog.cs#L20)):*
> ```Smalltalk
> 	/// <remarks>Unfortunately we haven't yet found a good way to make a Form with its
> 	/// title rendered in HTML draggable.</remarks>
> 	public partial class TeamCollectionDialog : Form
> ```

I think maybe this version in the Tests was a mistake?
And the real one didn't get the class comment. But see my comment there about making it a more generic class.
___
*[src/BloomTests/TeamCollection/TeamCollectionDialog.Designer.cs, line 4 at r1](https://reviewable.io/reviews/bloombooks/bloomdesktop/4243#-MUolgP16Ut63Q7t_p11:-MUolgP16Ut63Q7t_p12:b1z3xu8) ([raw file](https://github.com/bloombooks/bloomdesktop/blob/0e6b56dfb7ed8a9cebe1400125e3134cd527cdfb/src/BloomTests/TeamCollection/TeamCollectionDialog.Designer.cs#L4)):*
> ```Smalltalk
> namespace BloomTests.TeamCollection
> {
> 	partial class TeamCollectionDialog
> ```

See comment on other part of class.
___
*[src/BloomTests/TeamCollection/TeamCollectionMessageLogTests.cs, line 376 at r1](https://reviewable.io/reviews/bloombooks/bloomdesktop/4243#-MUp-DcZAX0j8cz7VsCt:-MUp-Dc_2G5XlRCLKPFG:b-fhqc3j) ([raw file](https://github.com/bloombooks/bloomdesktop/blob/0e6b56dfb7ed8a9cebe1400125e3134cd527cdfb/src/BloomTests/TeamCollection/TeamCollectionMessageLogTests.cs#L376)):*
> ```Smalltalk
> 			MakeClobberMessage();
> 			_messageLog.WriteMilestone(MessageAndMilestoneType.Reloaded);
> 			Assert.That(_messageLog.TeamCollectionStatus, Is.EqualTo(TeamCollectionStatus.Nominal));
> ```

This is a design question. And it should have come up when I was in the code but here it is...
It seems very likely the user could have an error state icon, not see it, close Bloom, and never know there was an error. We've decided not to have the user acknowledge errors, I guess?
___
*[src/BloomTests/TeamCollection/TeamCollectionMessageTests.cs, line 32 at r1](https://reviewable.io/reviews/bloombooks/bloomdesktop/4243#-MUp0kE16smyIlpvkPYS:-MUp0kE16smyIlpvkPYT:b-6cygh4) ([raw file](https://github.com/bloombooks/bloomdesktop/blob/0e6b56dfb7ed8a9cebe1400125e3134cd527cdfb/src/BloomTests/TeamCollection/TeamCollectionMessageTests.cs#L32)):*
> ```Smalltalk
> 			Assert.That(msg.When.Month, Is.EqualTo(6));
> 			Assert.That(msg.MessageType, Is.EqualTo(MessageAndMilestoneType.Reloaded));
> 			Assert.That(msg.Param1, Is.Null);
> ```

Why assert this is null without the other parts? Seems like this is the least likely to get incorrectly set since it is last.
___
*[src/BloomExe/TeamCollection/BookStatusChangeEventArgs.cs, line 12 at r1](https://reviewable.io/reviews/bloombooks/bloomdesktop/4243#-MUoqUfjBkGhDtGFOEQX:-MUoqUfjBkGhDtGFOEQY:b-vmtz0n) ([raw file](https://github.com/bloombooks/bloomdesktop/blob/0e6b56dfb7ed8a9cebe1400125e3134cd527cdfb/src/BloomExe/TeamCollection/BookStatusChangeEventArgs.cs#L12)):*
> ```Smalltalk
> 	}
> 
> 	public class BookStatusChangeEventArgs
> ```

I find myself wondering if this needs to somehow be more specific.
Seeing the name in other contexts, there is nothing to tie it to Team Collections at all.

BookTCStatusChangeEventArgs is an option, but I'm not sure we want to start using TC everywhere, either.

**Review by andrew-polk — CHANGES_REQUESTED** [review]
> Reviewed 1 of 32 files at r1, 12 of 13 files at r2.
> *[Reviewable](https://reviewable.io/reviews/bloombooks/bloomdesktop/4243)* status: all files reviewed, 9 unresolved discussions (waiting on @JohnThomson)

___
*[src/BloomExe/TeamCollection/TeamCollection.cs, line 905 at r1](https://reviewable.io/reviews/bloombooks/bloomdesktop/4243#-MUp98Nc2YJ1KmsY18P0:-MUp98Nc2YJ1KmsY18P1:b-c15s6b) ([raw file](https://github.com/bloombooks/bloomdesktop/blob/0e6b56dfb7ed8a9cebe1400125e3134cd527cdfb/src/BloomExe/TeamCollection/TeamCollection.cs#L905)):*
> ```
> l10nIdprefix
> ```
l10nIdSuffix?
___
*[src/BloomExe/MiscUI/ReactDialog.Designer.cs, line 54 at r2](https://reviewable.io/reviews/bloombooks/bloomdesktop/4243#-MUpAIbAAQT1g5oWhckf:-MUpAIbAAQT1g5oWhckg:b7qv501) ([raw file](https://github.com/bloombooks/bloomdesktop/blob/744a2d39e064b57bf3c0d6f294bdfcb361b2a430/src/BloomExe/MiscUI/ReactDialog.Designer.cs#L54)):*
> ```
> this.Text = "TeamCollectionDialog";
> ```
Oops. Missed one.
And actually the comment for this section, too.
___
*[src/BloomExe/TeamCollection/BookStatusChangeEventArgs.cs, line 12 at r1](https://reviewable.io/reviews/bloombooks/bloomdesktop/4243#-MUoqUfjBkGhDtGFOEQX:-MUp85co4-urQHzbgk69:b52u3td) ([raw file](https://github.com/bloombooks/bloomdesktop/blob/0e6b56dfb7ed8a9cebe1400125e3134cd527cdfb/src/BloomExe/TeamCollection/BookStatusChangeEventArgs.cs#L12)):*
<details><summary><i>Previously, andrew-polk wrote…</i></summary><blockquote>

I find myself wondering if this needs to somehow be more specific.
Seeing the name in other contexts, there is nothing to tie it to Team Collections at all.

BookTCStatusChangeEventArgs is an option, but I'm not sure we want to start using TC everywhere, either.
</blockquote></details>

But... if I'm right, we would now have BookStateChangeEventArgs and BookStatusChangeEventArgs.
Do we really intend to have both? If yes, they definitely need different names and helpful comments.

**Review by JohnThomson — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/bloombooks/bloomdesktop/4243)* status: 21 of 32 files reviewed, 10 unresolved discussions (waiting on @andrew-polk and @hatton)

___
*[src/BloomBrowserUI/teamCollection/TeamSettingsDialog.tsx, line 14 at r1](https://reviewable.io/reviews/bloombooks/bloomdesktop/4243#-MUojqGwEUIFIYZOzx_d:-MUp98lC9L1gIC5j-sgk:b-896fix) ([raw file](https://github.com/bloombooks/bloomdesktop/blob/0e6b56dfb7ed8a9cebe1400125e3134cd527cdfb/src/BloomBrowserUI/teamCollection/TeamSettingsDialog.tsx#L14)):*
<details><summary><i>Previously, andrew-polk wrote…</i></summary><blockquote>

> ```
> TeamCollection.DialogTitle
> ```
We already have TeamCollection.TeamCollection
</blockquote></details>

Done.
___
*[src/BloomExe/TeamCollection/NewBookEventArgs.cs, line 11 at r1](https://reviewable.io/reviews/bloombooks/bloomdesktop/4243#-MUox0RD5-uVCs5SXPSH:-MUpBhhH1rI8TQDNXfL7:b-4yymrj) ([raw file](https://github.com/bloombooks/bloomdesktop/blob/0e6b56dfb7ed8a9cebe1400125e3134cd527cdfb/src/BloomExe/TeamCollection/NewBookEventArgs.cs#L11)):*
<details><summary><i>Previously, andrew-polk wrote…</i></summary><blockquote>

> ```
> BookStateChangeEventArgs
> ```
BookStatusChangeEventArgs?
</blockquote></details>

We unfortunately wound up with both. I'm changing this one to BookRepoChangeEventArgs
___
*[src/BloomExe/TeamCollection/TeamCollection.cs, line 905 at r1](https://reviewable.io/reviews/bloombooks/bloomdesktop/4243#-MUp98Nc2YJ1KmsY18P0:-MUpC-TPDgzV73ybE4Q_:b-896fix) ([raw file](https://github.com/bloombooks/bloomdesktop/blob/0e6b56dfb7ed8a9cebe1400125e3134cd527cdfb/src/BloomExe/TeamCollection/TeamCollection.cs#L905)):*
<details><summary><i>Previously, andrew-polk wrote…</i></summary><blockquote>

> ```
> l10nIdprefix
> ```
l10nIdSuffix?
</blockquote></details>

Done.
___
*[src/BloomExe/TeamCollection/TeamCollectionApi.cs, line 71 at r1](https://reviewable.io/reviews/bloombooks/bloomdesktop/4243#-MUopVJT0KOr7P-n3En7:-MUpC1bN9-C2yytpYys_:bipefh8) ([raw file](https://github.com/bloombooks/bloomdesktop/blob/0e6b56dfb7ed8a9cebe1400125e3134cd527cdfb/src/BloomExe/TeamCollection/TeamCollectionApi.cs#L71)):*
<details><summary><i>Previously, andrew-polk wrote…</i></summary><blockquote>

Thinking out loud...
Since more and more things are moving toward react dialogs, etc., should we consider moving HandleCloseDialog and HandleReload to something outside of Team Collections?
</blockquote></details>

Moved to CommonApi
___
*[src/BloomExe/TeamCollection/TeamCollectionManager.cs, line 74 at r1](https://reviewable.io/reviews/bloombooks/bloomdesktop/4243#-MUp2bcVAwfZg7mcIpw1:-MUpCH3y2QooWlNxX_Ue:b-896fix) ([raw file](https://github.com/bloombooks/bloomdesktop/blob/0e6b56dfb7ed8a9cebe1400125e3134cd527cdfb/src/BloomExe/TeamCollection/TeamCollectionManager.cs#L74)):*
<details><summary><i>Previously, andrew-polk wrote…</i></summary><blockquote>

> ```
> var tcLogPath = GetTcLogPathFromLcPath(_localCollectionFolder);
> ```
never used
</blockquote></details>

Done.
___
*[src/BloomExe/TeamCollection/TeamCollectionMessage.cs, line 94 at r1](https://reviewable.io/reviews/bloombooks/bloomdesktop/4243#-MUotF-Q0GlaTB_lxHxw:-MUpCRWFBQxh94_cUmrU:b-896fix) ([raw file](https://github.com/bloombooks/bloomdesktop/blob/0e6b56dfb7ed8a9cebe1400125e3134cd527cdfb/src/BloomExe/TeamCollection/TeamCollectionMessage.cs#L94)):*
<details><summary><i>Previously, andrew-polk wrote…</i></summary><blockquote>

> ```
> Displayed error log
> ```
This gets displayed in non-error conditions, sometimes, doesn't it?
How about just "Displayed log"?
</blockquote></details>

Done.
___
*[src/BloomExe/TeamCollection/TeamCollectionMessage.cs, line 96 at r1](https://reviewable.io/reviews/bloombooks/bloomdesktop/4243#-MUotO0N5cfKeUh9BAst:-MUpCV3_8pAW_z5FFEuo:b-xqi6av) ([raw file](https://github.com/bloombooks/bloomdesktop/blob/0e6b56dfb7ed8a9cebe1400125e3134cd527cdfb/src/BloomExe/TeamCollection/TeamCollectionMessage.cs#L96)):*
<details><summary><i>Previously, andrew-polk wrote…</i></summary><blockquote>

> ```
> Repaired conflict
> ```
We didn't really repair anything did we?
How about "Displayed conflict notification"?
</blockquote></details>

Implementation is pending. The intent is that this milestone will only be added after repairing things.
___
*[src/BloomTests/TeamCollection/TeamCollectionDialog.cs, line 20 at r1](https://reviewable.io/reviews/bloombooks/bloomdesktop/4243#-MUolXdO6Fl8QqHLtXi9:-MUpCd5TBgO_Gb79BUDn:bbc098b) ([raw file](https://github.com/bloombooks/bloomdesktop/blob/0e6b56dfb7ed8a9cebe1400125e3134cd527cdfb/src/BloomTests/TeamCollection/TeamCollectionDialog.cs#L20)):*
<details><summary><i>Previously, andrew-polk wrote…</i></summary><blockquote>

I think maybe this version in the Tests was a mistake?
And the real one didn't get the class comment. But see my comment there about making it a more generic class.
</blockquote></details>

Removed, after moving comment and then rewording for ReactDialog
___
*[src/BloomTests/TeamCollection/TeamCollectionMessageLogTests.cs, line 376 at r1](https://reviewable.io/reviews/bloombooks/bloomdesktop/4243#-MUp-DcZAX0j8cz7VsCt:-MUpCqZYBT7ldZ1pOI0d:bnw6dob) ([raw file](https://github.com/bloombooks/bloomdesktop/blob/0e6b56dfb7ed8a9cebe1400125e3134cd527cdfb/src/BloomTests/TeamCollection/TeamCollectionMessageLogTests.cs#L376)):*
<details><summary><i>Previously, andrew-polk wrote…</i></summary><blockquote>

This is a design question. And it should have come up when I was in the code but here it is...
It seems very likely the user could have an error state icon, not see it, close Bloom, and never know there was an error. We've decided not to have the user acknowledge errors, I guess?
</blockquote></details>

One to raise with @hatton? I think the theory is that these problems will be very rare and the automatic fixes will often be good. If the user notices some changes were lost he can go back through the log. There is also some thought of a plan to encourage someone to pay attention to things that get put in Lost and Found. May be a case of "time will tell" whether errors getting lost is a problem.
___
*[src/BloomTests/TeamCollection/TeamCollectionMessageTests.cs, line 32 at r1](https://reviewable.io/reviews/bloombooks/bloomdesktop/4243#-MUp0kE16smyIlpvkPYS:-MUpDgVa786g7FZ6jJKZ:b-896fix) ([raw file](https://github.com/bloombooks/bloomdesktop/blob/0e6b56dfb7ed8a9cebe1400125e3134cd527cdfb/src/BloomTests/TeamCollection/TeamCollectionMessageTests.cs#L32)):*
<details><summary><i>Previously, andrew-polk wrote…</i></summary><blockquote>

Why assert this is null without the other parts? Seems like this is the least likely to get incorrectly set since it is last.
</blockquote></details>

Done.
___
*[src/BloomExe/MiscUI/ReactDialog.Designer.cs, line 54 at r2](https://reviewable.io/reviews/bloombooks/bloomdesktop/4243#-MUpAIbAAQT1g5oWhckf:-MUpEJqj3HwXJd6SMx0V:b-896fix) ([raw file](https://github.com/bloombooks/bloomdesktop/blob/744a2d39e064b57bf3c0d6f294bdfcb361b2a430/src/BloomExe/MiscUI/ReactDialog.Designer.cs#L54)):*
<details><summary><i>Previously, andrew-polk wrote…</i></summary><blockquote>

> ```
> this.Text = "TeamCollectionDialog";
> ```
Oops. Missed one.
And actually the comment for this section, too.
</blockquote></details>

Done.
___
*[src/BloomExe/TeamCollection/BookStatusChangeEventArgs.cs, line 12 at r1](https://reviewable.io/reviews/bloombooks/bloomdesktop/4243#-MUoqUfjBkGhDtGFOEQX:-MUpEPxw9FzwYLA-kWZ5:b-qj2iqp) ([raw file](https://github.com/bloombooks/bloomdesktop/blob/0e6b56dfb7ed8a9cebe1400125e3134cd527cdfb/src/BloomExe/TeamCollection/BookStatusChangeEventArgs.cs#L12)):*
<details><summary><i>Previously, andrew-polk wrote…</i></summary><blockquote>

But... if I'm right, we would now have BookStateChangeEventArgs and BookStatusChangeEventArgs.
Do we really intend to have both? If yes, they definitely need different names and helpful comments.
</blockquote></details>

I changed BookStateChange to BookRepoChange. Thinking YAGNI on making BookStatusChange more specific. See elsewhere.
___
*[src/BloomExe/TeamCollection/BookStatusChangeEventArgs.cs, line 12 at r2](https://reviewable.io/reviews/bloombooks/bloomdesktop/4243#-MUp9J7T48EPKCsNDJuF:-MUp9J7U8ryq9Qqp2_Dl:b-g7dum6) ([raw file](https://github.com/bloombooks/bloomdesktop/blob/744a2d39e064b57bf3c0d6f294bdfcb361b2a430/src/BloomExe/TeamCollection/BookStatusChangeEventArgs.cs#L12)):*
> ```Smalltalk
> 	}
> 
> 	public class BookStatusChangeEventArgs
> ```

Leaning towards YAGNI. With modern refactoring, it's not that hard to change to a more specific name if we one day have some other kind of BookStatus.
___
*[src/BloomExe/TeamCollection/TeamCollectionDialog.Designer.cs, line 4 at r1](https://reviewable.io/reviews/bloombooks/bloomdesktop/4243#-MUokszF47odgxy2oq3m:-MUp9AEgA7olzl02rM9a:b-896fix) ([raw file](https://github.com/bloombooks/bloomdesktop/blob/0e6b56dfb7ed8a9cebe1400125e3134cd527cdfb/src/BloomExe/TeamCollection/TeamCollectionDialog.Designer.cs#L4)):*
<details><summary><i>Previously, andrew-polk wrote…</i></summary><blockquote>

Unless I'm mistaken, this is just a form which holds a single ReactControl.
Should this be a more generic class? Like ReactDialog?
</blockquote></details>

Done.

**Review by andrew-polk — CHANGES_REQUESTED** [review]
> Reviewed 12 of 12 files at r3.
> *[Reviewable](https://reviewable.io/reviews/bloombooks/bloomdesktop/4243)* status: all files reviewed, 3 unresolved discussions (waiting on @hatton and @JohnThomson)

___
*[src/BloomExe/TeamCollection/BookRepoChangeEventArgs.cs, line 9 at r3](https://reviewable.io/reviews/bloombooks/bloomdesktop/4243#-MUpGXnIAt7Sc9Jgd-_I:-MUpGXnIAt7Sc9Jgd-_J:bkigqk) ([raw file](https://github.com/bloombooks/bloomdesktop/blob/955caa8bb8205379ba8b441eb45569b489b3818e/src/BloomExe/TeamCollection/BookRepoChangeEventArgs.cs#L9)):*
> ```Smalltalk
> namespace Bloom.TeamCollection
> {
> 	// Arguments for the BookStateChange event in ITeamRepo
> ```

This comment is obsolete.
I think it should be expanded enough that reading this comment and one for BookStatusChangeEventArgs would let a dev know which one to use.

Or probably it is better just to point to the event class and document well there. While we're at it, should these two event **arg** classes live in the same place as their corresponding event classes?

**Review by andrew-polk — COMMENTED** [review]
> Dismissed @JohnThomson from a discussion.
> *[Reviewable](https://reviewable.io/reviews/bloombooks/bloomdesktop/4243)* status: all files reviewed, 2 unresolved discussions (waiting on @hatton and @JohnThomson)

**Review by andrew-polk — COMMENTED** [review]
> Reviewed 1 of 1 files at r4.
> *[Reviewable](https://reviewable.io/reviews/bloombooks/bloomdesktop/4243)* status: all files reviewed, 2 unresolved discussions (waiting on @hatton and @JohnThomson)

**Review by andrew-polk — COMMENTED** [review]
> Reviewed 2 of 2 files at r5.
> *[Reviewable](https://reviewable.io/reviews/bloombooks/bloomdesktop/4243)* status: all files reviewed, 1 unresolved discussion (waiting on @hatton)

File: src/BloomExe/CollectionTab/LibraryListView.cs [hunk 1]
```diff
@@ -35,7 +35,7 @@ public partial class LibraryListView : UserControl
 
 		private readonly LibraryModel _model;
 		private readonly BookSelection _bookSelection;
-		private BookCheckoutStatusChangeEvent _tcCheckoutStatusChangeEvent;
+		private BookStatusChangeEvent _tcBookStatusChangeEvent;
```
Comment (hatton) [inline src/BloomExe/CollectionTab/LibraryListView.cs:38]: hello test comment

**Review by andrew-polk — COMMENTED** [review]
> Reviewed 13 of 13 files at r6.
> *[Reviewable](https://reviewable.io/reviews/bloombooks/bloomdesktop/4243)* status: all files reviewed, 2 unresolved discussions (waiting on @hatton and @JohnThomson)

**Review by JohnThomson — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/bloombooks/bloomdesktop/4243)* status: all files reviewed, 2 unresolved discussions (waiting on @hatton)

___
*[src/BloomExe/CollectionTab/LibraryListView.cs, line 38 at r5](https://reviewable.io/reviews/bloombooks/bloomdesktop/4243#-MUoZcOLCzF6DYPDp_l--r5-38:-MUtUqsqCkM9gmTZZTmX:b-qxgsy8) ([raw file](https://github.com/bloombooks/bloomdesktop/blob/4adafc5f14998f24bbb5b38568df791dc801dc54/src/BloomExe/CollectionTab/LibraryListView.cs#L38)):*
<details><summary><i>Previously, hatton (John Hatton) wrote…</i></summary><blockquote>

hello test comment
</blockquote></details>

hello back :-)
___
*[src/BloomExe/TeamCollection/BookRepoChangeEventArgs.cs, line 9 at r3](https://reviewable.io/reviews/bloombooks/bloomdesktop/4243#-MUpGXnIAt7Sc9Jgd-_I:-MUpJ9-gEj7g-fyAyVmA:b-896fix) ([raw file](https://github.com/bloombooks/bloomdesktop/blob/955caa8bb8205379ba8b441eb45569b489b3818e/src/BloomExe/TeamCollection/BookRepoChangeEventArgs.cs#L9)):*
<details><summary><i>Previously, andrew-polk wrote…</i></summary><blockquote>

This comment is obsolete.
I think it should be expanded enough that reading this comment and one for BookStatusChangeEventArgs would let a dev know which one to use.

Or probably it is better just to point to the event class and document well there. While we're at it, should these two event **arg** classes live in the same place as their corresponding event classes?
</blockquote></details>

Done.

**Pull request merged**